### PR TITLE
bootstrap tooltips

### PR DESCRIPTION
**Pull request by norgeous**
### Purpose
1. add an angular directive to apply js for bootstrap tooltips.
2. convert all existing title="" to use new tooltip directive.

#### normal tooltip example:
```<span title="hello">world</span>```

#### using new tooltip directive:
```<span tooltip data-original-title="hello">world</span>```

i believe bootstrap recommends using ```title``` attribute to populate the tooltips ([link] (http://www.w3schools.com/bootstrap/bootstrap_ref_js_tooltip.asp])), but by using ```data-original-title``` the tooltips live update without reapplying the js code, such as ```.tooltip('fixTitle')``` each time the content changes.

This method also works well with angular expressions:
```data-original-title="{{'Download Rate' | translate}}"```
This example provides a bootstrap tooltip saying 'Download Rate' that changes automatically when the language is updated.

### Testing

Developed with the override feature, by creating ```/gui``` inside the local ```AppData``` folder.
Worked in Firefox and SyncTrayzor with both ```default``` and ```dark``` themes.

### Screenshots
untitled

### CODE REVIEW

**Comment by calmh** [comment]
Neat!

**Comment by calmh** [comment]
@st-jenkins add to whitelist

**Comment by calmh** [comment]
What name would you like associated with yourself in the AUTHORS and credits?

**Comment by calmh** [comment]
In safari, this shows two tooltips (one of them with twice the content...) when the cell contents are truncated. I suspect this is just Safari being extra helpful as Chrome doesn't do this and I see nothing in the markup that ought to cause it.

screen shot 2016-03-10 at 08 11 35

Unless we see this somewhere else as well I'm inclined to ignore it as Safari weirdness.

**Comment by wweich** [comment]
The extra tooltip is probably the same as here: #2758
I couldn't reproduce that tooltip in Chrome or Firefox on Windows or Linux. So I'm pretty sure it is a safari only "feature" for truncated text.

**Comment by AudriusButkevicius** [comment]
Given it's title that causes that, should we change the attribute to something else?

**Comment by calmh** [comment]
So I tested this a bit closer on Safari now, and prior to this change I get the Safari-issued tooltip with the version in it just once:

screen shot 2016-03-10 at 09 33 39

This happens regardless of whether there is a `title="..."` attribute on the tag or not - so yes it seems based on the truncation, I guess it synthetically adds a title if there isn't one when it truncates the content, or some internal equivalent of that. 

What's odd is that the tooltip gets the contents duplicated with this patch... If we could get rid of that it would be nice.

**Comment by norgeous** [comment]
Hello All,
yes, I have just googled around for an answer to these Safari issues and can confirm;

1. Without the new bootstrap tooltips Safari always shows a native tooltip for truncated text with css ```text-overflow:ellipsis;```. I think I have read that this is against the w3c spec, and only Safari is doing it. There appears to be no way of disabling it; I have tried setting ```title=""``` empty titles on the tooltipped elements (no effect on these special Safari ellipsis tooltips) and setting css ```[tooltip]{pointer-events:none;}``` (this just disables interactions altogether, so no tooltips :S). Neither work suitably.

2. With the new bootstrap tooltips the native Safari tooltip displays the content twice for some reason. I tried ```<span tooltip data-original-title="1:{{folder.path}}">2:{{folder.path}}</span>``` and this happened:
untitled
The truncated text is shown first (2), followed by the ```data-original-title``` attribute on the next line (1).
How did Safari know about this custom attribute? must be a bug... idk

I can imagine some solution of putting the ellipsis'd text inside another span with ```cursor-events:none```, nested inside the tooltip element:
```
<td style="text-overflow:clip;">
    <span tooltip data-original-title="{{folder.path}}">
        <span style="text-overflow:ellipsis;">{{folder.path}}<span>
    </span>
</td>
```
I have limited testing capabilities for Safari Mac. But this seams to work. On the other hand, I don't want to add in a bunch more elements to solve the issue b/c it will quickly become confusing and some other programmer later on, may not know why its been done this way.

**TL;DR** I believe I can create a commit with it fixed if you would like (without inline styles)... but I will have to add in a extra few spans.

**Comment by calmh** [comment]
The safari fix worked, however now the tooltip pops up not from the text itself but from the center of the containing element (I guess?).

screen shot 2016-03-11 at 09 15 21

If that's the necessary tradeoff, I prefer the double tooltip on Safari.

**Comment by AudriusButkevicius** [comment]
It probably needs a css hack pull-right to right align the span.

**Comment by norgeous** [comment]
It appeared to me as if ```text-overflow``` cannot be applied to inline elements. Previously the ```text-overflow``` was on the table cell ```<td>```, but to fix the Safari issue I put the rule on the lowest element and turned it into a div (rather than a span).

I haven't done any further testing since last commit, but believe this new bug (tooltip not centered over text) is unavoidable. I only meant for the last commit to be an example of supressing the Safari bug, but it shouldn't be merged.

Personally, I think we (me?) should revert last commit and the Safari double bug stays.

**Comment by calmh** [comment]
Agreed. Do that :)

**Comment by calmh** [comment]
@norgeous A name for the credits?

**Comment by norgeous** [comment]
I have added my name and email to AUTHORS previously in this commit: https://github.com/syncthing/syncthing/pull/2828/commits/73df1696e02e5603c60d3005907f4a138f3429c0. Thanks!

**Comment by calmh** [comment]
Ah, but you're now committing from a different address. I'll add that, then.

**Comment by calmh** [comment]
In fact, two different addresses, so three in total... Some consistency here would be appreciated :)

**Comment by calmh** [comment]
@st-jenkins retest this please

**Comment by calmh** [comment]
@st-review merge this please

Use Bootstrap tooltips instead of plain title attributes

By using data-original-title the tooltips live update without reapplying the js code, such as .tooltip('fixTitle') each time the content changes. This method also works well with angular expressions:

    data-original-title="{{'Download Rate' | translate}}"

This example provides a bootstrap tooltip saying 'Download Rate' that changes automatically when the language is updated.

**Comment by st-review** [comment]
OK, merged as 78120bd989d78cc388273da067641771f3ddd9c4. Thanks, @norgeous!

**Comment by norgeous** [comment]
Thanks! Sorry for the inconsistent emails, this is my first time using git collaboratively.
